### PR TITLE
add: add model training code

### PR DESCRIPTION
**Pull request by moons98**
- add model training/inference code
- can be trained with various models / loss / dataset/ augmentation
- exp. is logged at wanDB, Notion

### CODE REVIEW

**Comment by moons98** [comment]
- Loss
    - CE, Label Smoothing, Focal Loss를 조합하여 사용 (class imbalance, over confident 문제 방지)
- Augmentation
    - Horizontal/Vertical Flip, Motion Blur, Hue_Saturation 등을 조절한 Hard Augmentation set 사용
- Optimizer, Scheduler
    - AdamW, ReduceLROnPlateau

- More Information
    - WanDB train log: https://wandb.ai/moons98/rsp_classification?nw=nwusermoons98
    - Exp. info: https://www.notion.so/Toy-Project-1b0cecb86d7180e38a64effc350e0f10?pvs=4
